addons/mail/models/mail_notification.py

```diff
@@ -114,7 +114,8 @@ def _filter_unimportant_notifications(notif):
                     or notif.res_partner_id.partner_share:
                 return True
             subtype = notif.mail_message_id.subtype_id
-            return not subtype or subtype.track_recipients
+            # sudo - access subtype to filter notifications is acceptable.
+            return not subtype or subtype.sudo().track_recipients
```

Comment:
It's acceptable indeed, but how comes it becomes an issue on this PR?

Reply:
Mostly for query count. Attr+sudo would cache the value. It's not the case anymore.

Reply:
What ?

Reply:
The order of read has changed in message to store, `Store.Attr("subtype_description", lambda m: m.subtype_id.sudo().description)` used to be run before `Store.Many(
                    "notification_ids",
value=lambda m: m.notification_ids._filtered_for_web_client(),`
This put the value in cache as sudo before the notification code.
`Store.One("subtype_id", ["description"], sudo=True),` is executed afterwards due to internals of `_get_records_data` of Store, the order has changed.
This means the notification code is now run first, which if not executed as sudo will add extra ACL queries that we didn't have before (and that we don't need).

Reply:
We are going to remove rights to access mail notification (even in stable) anyway, so I bet _filtered_for_web_client should be called as sudo anyway. Shouldn't that be better than adding sudo at that point ?

Reply:
It makes no difference in the scope of this PR, but yes if notification becomes technical (not a bad idea) then everything related to that should become sudo'ed. So we can have the sudo before reading the notif (for internal users only, as the current condition is set) instead of inside that method.